### ISSUE TITLE
Emoji library is out of date and needs overhaul/replacement

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#12741]
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### Description

Habitica uses [its own emoji library](https://github.com/HabitRPG/habitica-markdown-emoji) within its [Markdown implementation](https://github.com/HabitRPG/habitica-markdown). We parse strings like `:smiley_cat:` into image paths pointing to a collection of emoji stored on Amazon S3, which in turn come from [this collection](https://www.webfx.com/tools/emoji-cheat-sheet/) plus a few custom creations of our own.

All of these are wildly out of date vs. current emoji standards, and thus, users' expectations of how emoji should work and which ones are supported! Thus issue reports like #9128, requesting support for missing emoji; and #10927, where an attempted workaround by pasting raw UTF emoji confuses the module and leads to more broken image links.

Habitica's emoji handling needs an upgrade. Ideally, the solution should:

* Allow use of emoji from some recent standard, e.g. [the Unicode 13.1 set](https://unicode.org/emoji/charts/full-emoji-list.html)
* ...plus emoji of our own creation, like `:melior:`
* ...except those that we specifically want to disallow from the site, e.g. "middle finger"
* Parse shortcut text in a standard format, probably the CLDR Short Name of the emoji surrounded by `::` and with spaces replaced by underscores, e.g. `:ear_with_hearing_aid:`
* ...potentially with the ability to set additional aliases, such as for commonly used but wordy task-related emoji

This will require team and design input--choosing an image set for display, selecting disallowed emoji, etc.

**Comment by citrusella** [comment]
This might be better suited somewhere else but I figured I would ask it here since it's vaguely related to/questioning something one of the bullet points:

Is the third one (the disallowing) already supposed to be happening with some emoji (the way the custom emoji like melior in the previous bullet are already happening)? Or is it just desired whenever this happens? Because the middle finger emoji used as an example currently works on Habitica via both replacement of the Unicode version and via a specific but offensive `:code:` (viewable (commented out) in the source code of [this section of a subpage of my userpage on the wiki where I compiled all working emoji alt text](https://habitica.fandom.com/wiki/User:Citrusellaeditswikis/Emoji_cheat_sheet_attempt#Gestures_and_Body_Parts)) and I had been meaning to report that because it seemed unintended (probably through email because it seems like a bad idea to say "you can post middle fingers" right out in the open of the bug guild).

**Comment by SabreCat** [comment]
@citrusella: Nope, that's a wishlist item that I think the eventual solution should consider, not something we currently have going!

**Comment by JeppeKlitgaard** [comment]
Is there any update on this? In my view this is quite an important feature that is missing.

It might be worthwhile to add that Unicode emoji's currently have working behaviour in some places:
- ✔ Editing a task on web
- ✔ Task view on mobile
- ❌ Task view on web (gives 403 forbidden to do `emoji_name.png` not found on server)

**Comment by ofilipowicz** [comment]
To be honest, I think it should be split into two tasks. One is fixing current emojis which have not been working since many, many years already. Two is designing more emojis as mentioned by @SabreCat. 

Ticket no. 1 wouldn't require design team work and could be easily done by one developer. This way, we would at least have basic emojis working. Otherwise, we can wait for resources to do all design work indefinitely 😉

**Comment by citrusella** [comment]
I'm sure this is just me with a no-reading-comprehension-today brain, but when you say "fixing current emoji" do you mean "upgrading the library to a current standard" or "making sure emoji that should theoretically be in the current, non-updated library are working"?

I feel like I'm *sure* you probably mean the first one but wanted to ask because I couldn't wrap my head around what it meant as written.

**Comment by merwok** [comment]
I read it as your first interpretation.  First fix the emojis that exist now and work only partially, then do a broader rework or replacement of the whole library.

**Comment by CuriousMagpie** [comment]
I will be working closely with the rest of the team to determine the best approach for this issue. I know it's been a problem for a long time, but while this is something I am going to be working on, I am not actively working on it at this time.

**Comment by dotMavriQ** [comment]
Yeah, I used the "Stuffed Flatbread Emoji" for one of my Habits and it won't render properly in web browsers, so seemingly this is still an issue. [🥙](https://emojipedia.org/stuffed-flatbread/)

`<img class="habitica-emoji" style="height: 1.5em; width: 1.5em" src="https://s3.amazonaws.com/habitica-assets/cdn/emoji/stuffed_flatbread.png" alt="stuffed_flatbread" draggable="false">`

**Comment by dospolov** [comment]
I see that's still an issue
<img width="281" alt="image" src="https://github.com/HabitRPG/habitica/assets/670682/77d7ea1b-3542-4a7b-bca1-2ab9cb3b49f5">
The icon set is pretty limited
<img width="329" alt="image" src="https://github.com/HabitRPG/habitica/assets/670682/bc344e54-765e-42d2-8ef9-b4acb7fb1658">

Can I contribute somehow to the assets?

**Comment by rilygreat** [comment]
> A new bug report from @Postiljon:
> Some emojis are not displayed correctly on the web application: 
> ![jrjXrAV](https://github.com/user-attachments/assets/539cb66f-05cd-44ec-b2b8-d84303a7ca3a)
> 
> They are displayed correctly when in Edit menu or on mobile.

**Comment by luaralulu** [comment]
I'm having same issue. Doesn't display on Task title on Web. Edit menu and mobile working fine.
<img width="344" alt="Screenshot 2024-09-02 at 10 28 30" src="https://github.com/user-attachments/assets/19f304d1-57a8-4b78-b3b0-d9dc565de2ea">

**Comment by CuriousMagpie** [comment]
As stated [here](https://github.com/HabitRPG/habitica/issues/12741#issuecomment-1206570465), the team is aware of the issue. This is currently on hold. Thanks.

**Comment by EDLLT** [comment]
![image](https://github.com/user-attachments/assets/c4a7c3f1-8024-4430-acca-4f5da98e7814)
Facing the same issue, hopefully it'd be fixed eventually

**Comment by rilygreat** [comment]
> A new bug report from @ Noxor
> 
> Emojis break when joined with hyperlink (and probably with other markdowns) I'm trying to add: ✍️ [Log breakfast](HYPERLINK) or :writing_hand: [Log breakfast](HYPERLINK) in a checklist, and a broken image appears

**Comment by dylphuts** [comment]
Has there been any progress on this issue?

**Comment by luaralulu** [comment]
> Has there been any progress on this issue?

Still an issue for me on the browser view.

**Comment by Tressley** [comment]
The emoji library was fixed in a recent update.
